src/jitter.rs

```diff
@@ -424,18 +422,28 @@ impl JitterRng {
 
     fn gen_entropy(&mut self) -> u64 {
         trace!("JitterRng: collecting entropy");
-        
-        // Prime `self.prev_time`, and run the noice sources to make sure the
+
+        // Prime `ec.prev_time`, and run the noice sources to make sure the
         // first loop round collects the expected entropy.
-        let _ = self.measure_jitter();
+        let mut ec = EcState {
+            prev_time: self.timer.get_nstime(),
+            last_delta: 0,
+            last_delta2: 0,
```

Comment:
Not really sure about these deltas — you're sure they don't need to be preserved?

Reply:
They are only necessary for the stuck test. Now we set `prev_time` when initializing `EcState`, and set it again together with `last_delta` in the first dummy round of `measure_jitter`. That dummy round was already part of the algorithm.

We are only missing the previous value of `last_delta2`, making the stuck test on the first real round of `measure_jitter` slightly less reliable. But using a `last_delta2` that could have been set seconds or minutes earlier was not all that reliable already anyway.

So I see no loss here, but it takes some thinking through...